src/java.base/share/classes/java/lang/StringCoding.java

```diff
@@ -97,29 +116,83 @@ public static int countPositives(byte[] ba, int off, int len) {
         return len;
     }
 
+    /**
+     * Encodes as many ISO-8859-1 codepoints as possible from the source byte
+     * array containing characters encoded in UTF-16, into the destination byte
+     * array, assuming that the encoding is ISO-8859-1 compatible.
+     *
+     * @apiNote
+     *
+     * {@code sa} denotes the {@code byte[]} backing a {@link String}. When
+     * {@linkplain String#COMPACT_STRINGS compact strings} are disabled, a
+     * {@code char} is always represented by 2 bytes, i.e.,
+     * encoded in UTF-16. When enabled, if the content is ISO-8859-1, a
+     * {@code char} is represented by 1 byte; otherwise again by 2 bytes.
+     * <p>
+     * This method assumes that {@code sa} is encoded in UTF-16, and hence,
+     * each {@code char} maps to 2 bytes.
+     * </p><p>
+     * {@code sp} is encoded in ISO-8859-1. There each {@code byte} corresponds
+     * to a {@code char}.
+     * </p>
+     *
+     * @param sa the source byte array containing characters encoded in UTF-16
+     * @param sp the index of the <em>byte (not character!)</em> from the source array to start reading from
+     * @param da the target byte array
+     * @param dp the index of the target array to start writing to
+     * @param len the total number of <em>characters (not bytes!)</em> to be encoded
+     * @return the total number of <em>characters (not bytes!)</em> successfully encoded
+     * @throws NullPointerException if any of the provided arrays is null
+     * @throws ArrayIndexOutOfBoundsException if any of the provided sub-ranges are
+     *         {@linkplain Preconditions#checkFromIndexSize(int, int, int, BiFunction) out of bounds}
+     */
+    static int encodeISOArray(byte[] sa, int sp, byte[] da, int dp, int len) {
+        checkFromIndexSize(sp, len << 1, requireNonNull(sa, "sa").length, AIOOBE_FORMATTER);
+        checkFromIndexSize(dp, len, requireNonNull(da, "da").length, AIOOBE_FORMATTER);
+        return encodeISOArray0(sa, sp, da, dp, len);
+    }
+
     @IntrinsicCandidate
-    public static int implEncodeISOArray(byte[] sa, int sp,
-                                         byte[] da, int dp, int len) {
+    private static int encodeISOArray0(byte[] sa, int sp, byte[] da, int dp, int len) {
         int i = 0;
         for (; i < len; i++) {
             char c = StringUTF16.getChar(sa, sp++);
             if (c > '\u00FF')
                 break;
-            da[dp++] = (byte)c;
+            da[dp++] = (byte) c;
```

Comment:
Revert: Omit the space after (byte); there many more usages of "(byte)c" and this file does not need to drift from the prevailing style in StringUTF16.java AbstractStringBuilder, etc.
Here and at line 195:

Reply:
I've actually made these changes to match the code shared in intrinsics, e.g., `macroAssembler_x86.cpp`. Reverted them in 2672f7c1ada as you requested.